lib/dragonrealms/commons/slackbot.rb

```diff
@@ -1,21 +1,73 @@
 module Lich
   module DragonRealms
     class SlackBot
+      class Error < StandardError; end
+      class NetworkError < Error; end
+      class ApiError < Error; end
+
+      class ThrottlingError < ApiError
+        attr_reader :retry_after
+
+        def initialize(msg, retry_after = nil)
+          super(msg)
+          @retry_after = retry_after
+        end
+      end
+
       def initialize
         @api_url = 'https://slack.com/api/'
-        @lnet = (Script.running + Script.hidden).find { |val| val.name == 'lnet' }
-        find_token unless authed?(UserVars.slack_token)
 
-        params = { 'token' => UserVars.slack_token }
-        res = post('users.list', params)
-        @users_list = JSON.parse(res.body)
+        unless authed?(UserVars.slack_token)
+          unless Script.running?('lnet') && lnet_connected?
+            start_script('lnet') unless Script.running?('lnet')
+            wait_time = 30
+            start_time = Time.now
+            until lnet_connected?
+              if (Time.now - start_time) > wait_time
+                raise Error, "lnet did not connect within #{wait_time} seconds."
+              end
+              pause 1
+            end
```

Comment:
I haven't messed with slackbot/lnetbot stuff before, but depending on how Slackbot is called/initialized, would this cause main Lich thread to be blocked until lnet starts up?